eng/scripts/Language-Settings.ps1

```diff
@@ -6,27 +6,43 @@ $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/r
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=python%2F&delimiter=%2F"
 
 function Get-python-PackageInfoFromRepo  ($pkgPath, $serviceDirectory, $pkgName)
-{
-  pip install packaging==20.4 -q -I
-  $pkgName = $pkgName.Replace('_', '-')
+{  
+  $packageName = $pkgName.Replace('_', '-')
+  $pkgDirName = Split-Path $pkgPath -Leaf
+  if ($pkgDirName -ne $packageName)
+  {
+    # Common code triggers this function against each directory but we can skip if it doesn't match package name
+    return $null
+  }
+
   if (Test-Path (Join-Path $pkgPath "setup.py"))
   {
     $setupLocation = $pkgPath.Replace('\','/')
     pushd $RepoRoot
-    $setupProps = (python -c "import sys; import os; sys.path.append(os.path.join('scripts', 'devops_tasks')); from common_tasks import get_package_properties; obj=get_package_properties('$setupLocation'); print('{0},{1},{2}'.format(obj[0], obj[1], obj[2]));") -split ","
+    $setupProps = $null
+    try{
+      pip install packaging==20.4 -q -I
```

Comment:
Did you intend to move this out so its only installed once?

Reply:
Nope. It attempts to install for every package as opposed to every iteration. But we can move this completely outside too.

Reply:
I think an issue in that approach will be  additional constraint to have python installed if anyone wants to use any other functions in this script. For now, I will leave as it is since this change alone has improved performance a lot.